declarations/WebpackOptions.d.ts

```diff
@@ -2853,6 +2857,10 @@ export interface AssetResourceGeneratorOptions {
 	 * Specifies the filename template of output files on disk. You must **not** specify an absolute path here, but the path may contain folders separated by '/'! The specified path is joined with the value of the 'output.path' option to determine the location on disk.
 	 */
 	filename?: FilenameTemplate;
+	/**
+	 * Keep any source maps generated for assets by loaders.
+	 */
+	keepSourceMap?: boolean;
```

Comment:
maybe just `binary: boolean`, so if we will have more types (it is possible right now using plugin) we can allow handle source maps, I find out it more clear name and handle more possible cases, i.e.

```
this.generatorOptions.binary || this.binary ? asBuffer(source) : asString(source)  
```